cypress-io/cypress
# feat: Auto-focus and select file name for create new spec modal

- Closes #21865 

### User facing changelog


Opening the create spec modal will automatically focus the input and select the file name of the new spec

### Additional details


This change is an improvement to the UX for creating a new spec. Most of the time, the user will probably want to edit the file name to override the default. By auto-focusing the input and selecting the part of the file path that the user will most likely want to edit, we can speed up the spec creation process resulting in a more convenient UX. See related issue for more info.

### Steps to test


- Open the spec list for an application (e2e or ct)
- Click the "+ New Spec" button in the top-right corner
- Verify that when the modal opens, the text input is focused and the file name in the path is selected for you to edit

### How has the user experience changed?


**After**
![image](https://user-images.githubusercontent.com/14275198/173374986-1cf78b72-5bd8-422d-b377-d851126ffd7d.png)

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
